integration-tests/src/test/scala/chisel3/std/tilelink/TLMem.scala

```diff
@@ -0,0 +1,104 @@
+package chisel3.std.tilelink
+
+import chisel3._
+import chisel3.experimental.{SerializableModule, SerializableModuleParameter}
+import org.scalatest.flatspec.AnyFlatSpec
+
+/**
+  * Parameter for TLRam
+  */
+case class TLRamParameter(
+  busParam: TLBundleParameter)
+    extends SerializableModuleParameter {
+  require(!busParam.isTLC, "TLRam does not support TL-C")
+}
+
+/**
+  * A simple RAM attached to TileLink bus
+  */
+class TLRam(val parameter: TLRamParameter) extends Module with SerializableModule[TLRamParameter] {
+  val busIO: TLBundle = IO(new TLBundle(parameter.busParam))
+
+  val addrBits: Int = parameter.busParam.a.addressWidth
+  val dataBits: Int = parameter.busParam.a.dataWidth
+
+  private val mem = SyncReadMem(BigInt(1) << addrBits, Vec(dataBits / 8, UInt(8.W)))
+
+  private val a_read = Wire(Bool())
+  private val a_wen = Wire(Bool())
+  private val a_wdata = Wire(Vec(dataBits / 8, UInt(8.W)))
+  private val a_mask = Wire(Vec(dataBits / 8, Bool()))
+  private val a_addr = Wire(UInt(addrBits.W))
+  private val a_ready = Wire(Bool())
+
+  private val d_mem_out = Reg(UInt(dataBits.W))
+  private val d_resp = Reg(UInt(OpCode.width))
+  private val d_source = Reg(UInt(parameter.busParam.d.sourceWidth.W))
+  private val d_size = Reg(UInt(parameter.busParam.d.sizeWidth.W))
+  private val d_valid = RegInit(false.B)
```

Comment:
no need `private` here. and basically this is a bad design choice: since these binding cannot be accessed by XMR.